src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -329,10 +369,19 @@ public static string LocalSlotName
                 functionType.ToString(),
                 ExtensionVersion,
                 IsReplay: isReplay);
+            if (this.ShouldReplayEventsBeLogged(false))
+            {
+                this.logger.LogInformation(
+                    "{instanceId}: Function '{functionName} ({functionType})' was resumed by a timer scheduled for '{expirationTime}'. IsReplay: {isReplay}. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}. SequenceNumber: {sequenceNumber}.",
+                    instanceId, functionName, functionType, expirationTimeString, isReplay, FunctionState.TimerExpired,
+                    hubName, LocalAppName, LocalSlotName, ExtensionVersion, this.sequenceNumber++);
+            }
+        }
 
-            this.logger.LogInformation(
-                "{instanceId}: Function '{functionName} ({functionType})' was resumed by a timer scheduled for '{expirationTime}'. IsReplay: {isReplay}. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}. SequenceNumber: {sequenceNumber}.",
-                instanceId, functionName, functionType, expirationTimeString, isReplay, FunctionState.TimerExpired, hubName, LocalAppName, LocalSlotName, ExtensionVersion, this.sequenceNumber++);
+        private bool ShouldReplayEventsBeLogged(bool isReplay)
+        {
+            return this.durableTaskExtension.LogReplayEvents ||
+                   (!this.durableTaskExtension.LogReplayEvents && !isReplay);
```

Comment:
This condition seems a bit redundant. I think it's more concise to say `this.durableTaskExtension.LogReplayEvents || !isReplay`.